### ISSUE TITLE
Incorrect "unused value" warning on list with maps

### ISSUE THREAD

**Issue by kimmoahola** [clj-kondo/clj-kondo#2369]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

2024.08.01

**problem**

Maps inside a list causes the unused value warning for the maps.

**repro**

```
(ns f)

(fn [] '({:a 1} {:b 2} {:c 3}))
```

```
f.clj:3:10: warning: Unused value
f.clj:3:17: warning: Unused value
```

Just to note that `(fn [] '(1 2 3))` does not produce warnings.

**expected behavior**

No warnings produced for that line.
